python/paddle/fluid/dygraph/dygraph_to_static/utils.py

```diff
@@ -1195,3 +1196,39 @@ def _is_wrapped(f):
         unwrapped_f = unwrapped_f.__wrapped__
 
     return unwrapped_f
+
+
+def input_specs_compatible(src_input_specs, other_input_specs):
+    """
+    Returns True if the two input specs are compatible, otherwise False.
+
+    args:
+        src_input_spec (list[InputSpec]|tuple(InputSpec)): list/tuple of
+            paddle.static.InputSpec
+        other_input_spec (list[InputSpec]|tuple(InputSpec)): list/tuple of
+            paddle.static.InputSpec
+    """
+    len_specs = len(src_input_specs)
```

Comment:
Maybe it's better to check that whether `src_input_spec`  and `other_input_specs ` have the same nested structure first. Because the items in src_input_specs may be InputSpec, dict, list, etc.

Reply:
Good! I have followed @Aurelius84 's suggestion to add `flatten` for inputs to make sure those inputs are list of InputSpecs.